elm/View.elm

```diff
@@ -71,39 +70,67 @@ root model =
     }
 
 
-header :
-    Wallet
-    -> String
-    -> Element Msg
+header : Wallet -> String -> Element Msg
 header wallet searchInput =
-    Element.el
-        [ Element.width Element.fill
-        , Element.height <| Element.px 80
-        , Element.Background.color EH.black
-        , EH.moveToFront
-        , whiteGlowAttribute
-        , Element.paddingXY 0 15
+    [ Element.image
+        [ height <| px 50
         ]
-    <|
-        Element.row
-            [ Element.paddingXY 100 0
-            , Element.Font.color Theme.orange
+        { src = "img/smokesignal-logo-horizontal.svg"
+        , description = "smokesignal logo"
+        }
+    , Input.text
+        [ fill |> Element.maximum 350 |> width
+        , Background.color black
+        ]
+        { onChange = always ClickHappened
+        , label = Input.labelHidden ""
+        , placeholder =
+            "Search . . ."
+                |> text
+                |> Input.placeholder []
+                |> Just
+        , text = searchInput
+        }
+    , [ Element.image
+            [ height <| px 50
             ]
-            [ Element.image
-                [ Element.height <| Element.px 50
-                ]
-                { src = "img/smokesignal-logo-horizontal.svg"
-                , description = "smokesignal logo"
-                }
-            , Element.row
-                [ Element.height Element.fill
-                , Element.width <| Element.px 0
-                , Element.Border.glow
-                    (Element.rgba 1 1 1 0.4)
-                    40
-                , Element.moveLeft 50
-                ]
-                []
+            { src = "/img/share.svg"
+            , description = ""
+            }
+      , Element.image
+            [ height <| px 50
+            ]
+            { src = "/img/info.svg"
+            , description = ""
+            }
+      , Element.image
+            [ height <| px 50
+            ]
+            { src = "/img/foundry-icon.svg"
+            , description = ""
+            }
+      , "Login"
+            |> text
+            |> el []
+      , "Sign Up"
+            |> text
+            |> el []
+      ]
+        |> row [ width fill, spaceEvenly ]
+    ]
+        |> row
```

Comment:
Really into the changes you're making generally, but this is more confusing to me. When reading view code, I'd prefer to know *first* whether something is a row or column at its base, then get to the list of items.